<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Scope" data-toc-modified-id="Scope-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Scope</a></span></li><li><span><a href="#Importing-libraries" data-toc-modified-id="Importing-libraries-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Importing libraries</a></span></li><li><span><a href="#Read-in-files" data-toc-modified-id="Read-in-files-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Read in files</a></span></li><li><span><a href="#Cleaning-process" data-toc-modified-id="Cleaning-process-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Cleaning process</a></span><ul class="toc-item"><li><span><a href="#Covers" data-toc-modified-id="Covers-4.1"><span class="toc-item-num">4.1&nbsp;&nbsp;</span>Covers</a></span></li><li><span><a href="#Skits" data-toc-modified-id="Skits-4.2"><span class="toc-item-num">4.2&nbsp;&nbsp;</span>Skits</a></span></li><li><span><a href="#Remixes" data-toc-modified-id="Remixes-4.3"><span class="toc-item-num">4.3&nbsp;&nbsp;</span>Remixes</a></span></li><li><span><a href="#Duplicates" data-toc-modified-id="Duplicates-4.4"><span class="toc-item-num">4.4&nbsp;&nbsp;</span>Duplicates</a></span></li><li><span><a href="#Instrumentals" data-toc-modified-id="Instrumentals-4.5"><span class="toc-item-num">4.5&nbsp;&nbsp;</span>Instrumentals</a></span></li><li><span><a href="#Lyrics" data-toc-modified-id="Lyrics-4.6"><span class="toc-item-num">4.6&nbsp;&nbsp;</span>Lyrics</a></span></li><li><span><a href="#Adlibs-and-verse-indication" data-toc-modified-id="Adlibs-and-verse-indication-4.7"><span class="toc-item-num">4.7&nbsp;&nbsp;</span>Adlibs and verse indication</a></span></li><li><span><a href="#Removing-unwanted-phrases" data-toc-modified-id="Removing-unwanted-phrases-4.8"><span class="toc-item-num">4.8&nbsp;&nbsp;</span>Removing unwanted phrases</a></span></li><li><span><a href="#Non-english-words" data-toc-modified-id="Non-english-words-4.9"><span class="toc-item-num">4.9&nbsp;&nbsp;</span>Non-english words</a></span></li><li><span><a href="#Punctuation" data-toc-modified-id="Punctuation-4.10"><span class="toc-item-num">4.10&nbsp;&nbsp;</span>Punctuation</a></span></li><li><span><a href="#Removing-cuss-words" data-toc-modified-id="Removing-cuss-words-4.11"><span class="toc-item-num">4.11&nbsp;&nbsp;</span>Removing cuss words</a></span></li><li><span><a href="#Blank-spaces-and-lines" data-toc-modified-id="Blank-spaces-and-lines-4.12"><span class="toc-item-num">4.12&nbsp;&nbsp;</span>Blank spaces and lines</a></span></li></ul></li><li><span><a href="#Exporting-dataframe-into-csv" data-toc-modified-id="Exporting-dataframe-into-csv-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Exporting dataframe into csv</a></span></li></ul></div>

### Scope
To clean the lyrics column in the data frame.

Note: Checking/testing lines of code may be commented out just to hide the cuss words.

### Importing libraries

In [2]:
import pandas as pd
import numpy as np
import re
import nltk

### Read in files

In [3]:
lyrical_df = pd.read_csv("lyrical_df.csv")

In [4]:
# Check format
lyrical_df

lyrics  \
0     Feels like we're on the edge right now\nI wish...   
1     "Hey, Nate, how's life?"\nI don’t know, it's a...   
2     Yeah, when I grow up, you know what I wanna be...   
3     I heard you told your friends that I'm just no...   
4     Clicks, clicks, clicks, they'll do anything to...   
...                                                 ...   
4023  [Verse 1]\nToken goin' in, you can call the be...   
4024  [Chorus]\nBack it up, back it up\nFlow too cra...   
4025  [Verse: Token]\nSeptember 24th 1998 was the ti...   
4026  [Intro]\nUh\n\n[Chorus]\nIt's summer break, ye...   
4027  [Intro]\nKato production\n\n[Chorus]\nI don't ...   

                                                   song artist  
0     "Let You Down" by NF:\n    Feels like we're on...     nf  
1     "The Search" by NF:\n    "Hey, Nate, how's lif...     nf  
2     "When I Grow Up" by NF:\n    Yeah, when I grow...     nf  
3     "Lie" by NF:\n    I heard you told your friend...     nf  
4     "PAID MY DUES" by NF:\n    Clicks, clicks, cli...     nf  
...                                                 ...    ...  
4023  "Left Somethin’" by Token:\n    [Verse 1]\n   ...  token  
4024  "Back It Up*" by Token:\n    [Chorus]\n    Bac...  token  
4025  "Jurassic Cypher" by Token:\n    [Verse: Token...  token  
4026  "Summer Break" by Token:\n    [Intro]\n    Uh\...  token  
4027  "Yellow Tape" by Token:\n    [Intro]\n    Kato...  token  

[4028 rows x 3 columns]

In [5]:
# Check artists
lyrical_df["artist"].value_counts()

eminem               546
TI                   398
kendrick lamar       330
jcole                280
bigsean              261
nipsey hussle        255
icecube              251
mfdoom               196
hopsin               178
chance the rapper    177
mosdef               166
denzelcurry          155
joyner lucas         132
nf                   131
bigl                 128
token                 96
YBNCordae             91
natedogg              91
lil dicky             90
macklemore            76
Name: artist, dtype: int64

Note: I have a feeling that the the neural network might learn have a bias towards the style of an artist whose songs were extracted the most. Ideally, I could potentially extract the same number of songs per artist to even out the bias. On top of that, I could select the most popular songs for the network to study from. BUT, I personally would like the network to get familiar with Eminem's style. Mainly because of his word play and his rhyming scheme. 

In [6]:
# Check
lyrical_df

lyrics  \
0     Feels like we're on the edge right now\nI wish...   
1     "Hey, Nate, how's life?"\nI don’t know, it's a...   
2     Yeah, when I grow up, you know what I wanna be...   
3     I heard you told your friends that I'm just no...   
4     Clicks, clicks, clicks, they'll do anything to...   
...                                                 ...   
4023  [Verse 1]\nToken goin' in, you can call the be...   
4024  [Chorus]\nBack it up, back it up\nFlow too cra...   
4025  [Verse: Token]\nSeptember 24th 1998 was the ti...   
4026  [Intro]\nUh\n\n[Chorus]\nIt's summer break, ye...   
4027  [Intro]\nKato production\n\n[Chorus]\nI don't ...   

                                                   song artist  
0     "Let You Down" by NF:\n    Feels like we're on...     nf  
1     "The Search" by NF:\n    "Hey, Nate, how's lif...     nf  
2     "When I Grow Up" by NF:\n    Yeah, when I grow...     nf  
3     "Lie" by NF:\n    I heard you told your friend...     nf  
4     "PAID MY DUES" by NF:\n    Clicks, clicks, cli...     nf  
...                                                 ...    ...  
4023  "Left Somethin’" by Token:\n    [Verse 1]\n   ...  token  
4024  "Back It Up*" by Token:\n    [Chorus]\n    Bac...  token  
4025  "Jurassic Cypher" by Token:\n    [Verse: Token...  token  
4026  "Summer Break" by Token:\n    [Intro]\n    Uh\...  token  
4027  "Yellow Tape" by Token:\n    [Intro]\n    Kato...  token  

[4028 rows x 3 columns]

In [7]:
# Checking data type - lyrics
lyrical_df["lyrics"].dtype

dtype('O')

Need to make sure the column `lyrics` always has a string format so that I can perform regex operations if needed. 

In [8]:
# Converting into string
lyrical_df["lyrics"] = lyrical_df["lyrics"].astype(str)

### Cleaning process

In [9]:
lyrical_df.shape

(4028, 3)

Starting off with 4028 songs.

#### Covers

In [10]:
lyrical_df[lyrical_df["song"].str.contains("Cover")]

lyrics  \
227   I can tell you’re a real cool girl\nI mean it ...   
1186  Cover Art:\nChance the Rapper - "JUICE"\nChanc...   
2527  J. Cole - Born Sinner (Standard Cover)\nJ. Col...   
2566  J. Cole - Cole World: The Sideline Story\n5Emb...   
2929  Writer At WarKendrick Lamar in his own words.A...   

                                                   song             artist  
227   "Cover Up" by Hopsin:\n    I can tell you’re a...             hopsin  
1186  "Acid Rap [Artwork]" by Chance the Rapper:\n  ...  chance the rapper  
2527  "Born Sinner: Album Covers" by J. Cole:\n    J...              jcole  
2566  "Cole World: The Sideline Story Album Cover" b...              jcole  
2929  "Writer At War: Kendrick Lamar’s XXL Cover Sto...     kendrick lamar

In [11]:
lyrical_df[lyrical_df["song"].str.contains("cover")]

lyrics  \
152   Undercover Prodigy\nCome on!\n\nLook, I ain't ...   
1756  Hi! Original Bad Boy on the case, cover your f...   
1892                        EmbedShare URLCopyEmbedCopy   
2855  I want to describe my discovery in the simples...   

                                                   song          artist  
152   "Witch Doctor" by Hopsin:\n    Undercover Prod...          hopsin  
1756  "Any Man (Fucking Crazy)" by Eminem:\n    Hi! ...          eminem  
1892  "Recovery [Booklet]" by Eminem:\n    EmbedShar...          eminem  
2855  "Determined" by Kendrick Lamar:\n    I want to...  kendrick lamar

TALK ABOUT COVERS ABOVE. DROP IF FOUND

#### Skits

In [12]:
lyrical_df[lyrical_df["song"].str.contains("skit")]

Empty DataFrame
Columns: [lyrics, song, artist]
Index: []

In [13]:
lyrical_df[lyrical_df["song"].str.contains("Skit")]

lyrics  \
139   Man, rap today fucking sucks bad, I don't give...   
219   (H-H-H-Hot 92. 4, Where Hip-Hop lives)\n— Ladi...   
240   *Phone rings*\nHopsin: What do you want?\nAlyc...   
264   Yo, what's up everybody? Normally, I would sta...   
487   Hey, hey, hey come on in\n\nHey Baby\n\nHey, w...   
...                                                 ...   
3291  There's still DOOM, what of him?\n\nWhat of DO...   
3653  [Skit]\nGuard: Right this way, ma'am. Lockdown...   
3656  [Skit]\nPriest: Heavenly Father, I bow before ...   
3995  [Skit: Token, Mom, & Mailman]\nComing. Sorry t...   
3999  [Skit: Token]\nYo, I think that was the one, m...   

                                                   song    artist  
139   "No Words (Skit)" by Hopsin:\n    Man, rap tod...    hopsin  
219   "No Words 2 (Skit)" by Hopsin:\n    (H-H-H-Hot...    hopsin  
240   "Cute in a Suit (Skit)" by Hopsin:\n    *Phone...    hopsin  
264   "Intro - Skit" by Hopsin:\n    Yo, what's up e...    hopsin  
487   "The Breakup (Skit)" by T.I.:\n    Hey, hey, h...        TI  
...                                                 ...       ...  
3291  "The Mystery of Doom (Skit)" by MF DOOM:\n    ...    mfdoom  
3653  "Intro to G-Funk (Comm. 1)" by Nate Dogg:\n   ...  natedogg  
3656  "Last Prayer (Comm. 2)" by Nate Dogg:\n    [Sk...  natedogg  
3995  "Mail (Outro)" by Token:\n    [Skit: Token, Mo...     token  
3999  "Shyea" by Token:\n    [Skit: Token]\n    Yo, ...     token  

[84 rows x 3 columns]

Dropping songs with "Skit" written in them. 

In [14]:
# Removing Skits
lyrical_df = lyrical_df[~lyrical_df["song"].str.contains("Skit")]

#### Remixes

In [15]:
# lyrical_df[lyrical_df["song"].str.contains("remix")]

In [16]:
# Check
# lyrical_df["song"][613]

In [17]:
# Check
# lyrical_df["lyrics"][613]

It just seems the song and lyrics column do have similarities. My focus will solely be on the lyrics column so I don't plan to drop anything.

#### Duplicates

In [18]:
# drop song duplicates
lyrical_df.drop_duplicates(subset="song", keep=False, inplace=True)
                      
# drop lyric duplicates
lyrical_df.drop_duplicates(subset="lyrics", keep=False, inplace=True)

C:\Users\gsagararatne\Anaconda3\envs\deeplearning\lib\site-packages\pandas\util\_decorators.py:311: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return func(*args, **kwargs)


#### Instrumentals

In [19]:
lyrical_df[lyrical_df["song"].str.contains("instrumental")]

lyrics  \
2137  {*ten seconds of instrumental to open*}\n + (I...   

                                                   song   artist  
2137  "Life in California" by Ice Cube:\n    {*ten s...  icecube

In [20]:
# Removing Skits
lyrical_df = lyrical_df[~lyrical_df["song"].str.contains("instrumental")]

In [21]:
# Check
# lyrical_df["song"][2137]

In [22]:
# Check
# lyrical_df["lyrics"][2137]

Will delete the row that contains information about the song "Life in California" by Ice Cube just to save the hassle of cleaning the lyrics.

#### Lyrics

In [23]:
lyrical_df[lyrical_df["lyrics"].str.contains("lyrics")]

lyrics  \
2     Yeah, when I grow up, you know what I wanna be...   
12    Insidious is blind inception\nWhat's reality w...   
16    Look\nSelf-awareness\nPride's a coat and yes, ...   
49    (Yeah!) I'm back, did anyone miss me?\nThey sa...   
71    I've been-I've been doin' this a long time. I ...   
...                                                 ...   
3975  [Verse]\nI met this girl at my show\nShe went ...   
3984  [Intro]\nNow recently a lot of people have ope...   
3985  [Intro]\nShyea!\n\n[Verse 1]\nCocky son of a s...   
4006  [Verse 1]: Token\nWreckless flow spittin' loco...   
4020  [Intro:]\nWho do you believe in?\nWho do you b...   

                                                   song artist  
2     "When I Grow Up" by NF:\n    Yeah, when I grow...     nf  
12    "Mansion" by NF:\n    Insidious is blind incep...     nf  
16    "LOST" by NF:\n    Look\n    Self-awareness\n ...     nf  
49    "Intro 2" by NF:\n    (Yeah!) I'm back, did an...     nf  
71    "Motivated" by NF:\n    I've been-I've been do...     nf  
...                                                 ...    ...  
3975  "Suitcase and a Passport" by Token:\n    [Vers...  token  
3984  "Mass Reform" by Token:\n    [Intro]\n    Now ...  token  
3985  "#BadMemory" by Token:\n    [Intro]\n    Shyea...  token  
4006  "Tnght Bugg’n Freestyle" by Token:\n    [Verse...  token  
4020  "Who Do U Believe In?" by Token:\n    [Intro:]...  token  

[124 rows x 3 columns]

In [24]:
# check
# lyrical_df["lyrics"][2]

In [25]:
# check
# lyrical_df["lyrics"][3975]

Randomly checked 2 lines. The word "Lyrics" seemed to be used in verses. Will keep them.

#### Adlibs and verse indication

Removing lyrics inside square brackets. Mainly because they can be adlibs

In [26]:
lyrical_df

lyrics  \
0     Feels like we're on the edge right now\nI wish...   
1     "Hey, Nate, how's life?"\nI don’t know, it's a...   
2     Yeah, when I grow up, you know what I wanna be...   
3     I heard you told your friends that I'm just no...   
4     Clicks, clicks, clicks, they'll do anything to...   
...                                                 ...   
4023  [Verse 1]\nToken goin' in, you can call the be...   
4024  [Chorus]\nBack it up, back it up\nFlow too cra...   
4025  [Verse: Token]\nSeptember 24th 1998 was the ti...   
4026  [Intro]\nUh\n\n[Chorus]\nIt's summer break, ye...   
4027  [Intro]\nKato production\n\n[Chorus]\nI don't ...   

                                                   song artist  
0     "Let You Down" by NF:\n    Feels like we're on...     nf  
1     "The Search" by NF:\n    "Hey, Nate, how's lif...     nf  
2     "When I Grow Up" by NF:\n    Yeah, when I grow...     nf  
3     "Lie" by NF:\n    I heard you told your friend...     nf  
4     "PAID MY DUES" by NF:\n    Clicks, clicks, cli...     nf  
...                                                 ...    ...  
4023  "Left Somethin’" by Token:\n    [Verse 1]\n   ...  token  
4024  "Back It Up*" by Token:\n    [Chorus]\n    Bac...  token  
4025  "Jurassic Cypher" by Token:\n    [Verse: Token...  token  
4026  "Summer Break" by Token:\n    [Intro]\n    Uh\...  token  
4027  "Yellow Tape" by Token:\n    [Intro]\n    Kato...  token  

[3768 rows x 3 columns]

In [27]:
# Adlibs
lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace(r"\((.*?)\)","")

<ipython-input-27-02eb47c9cd01>:2: FutureWarning: The default value of regex will change from True to False in a future version.
  lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace(r"\((.*?)\)","")


In [28]:
# verse indications
lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace("\[.*?\]","")

<ipython-input-28-0d36b7672fbe>:2: FutureWarning: The default value of regex will change from True to False in a future version.
  lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace("\[.*?\]","")


In [29]:
lyrical_df

lyrics  \
0     Feels like we're on the edge right now\nI wish...   
1     "Hey, Nate, how's life?"\nI don’t know, it's a...   
2     Yeah, when I grow up, you know what I wanna be...   
3     I heard you told your friends that I'm just no...   
4     Clicks, clicks, clicks, they'll do anything to...   
...                                                 ...   
4023  \nToken goin' in, you can call the beat a slot...   
4024  \nBack it up, back it up\nFlow too crack I'm c...   
4025  \nSeptember 24th 1998 was the time that Satan ...   
4026  \nUh\n\n\nIt's summer break, yeah it's summer ...   
4027  \nKato production\n\n\nI don't see no competit...   

                                                   song artist  
0     "Let You Down" by NF:\n    Feels like we're on...     nf  
1     "The Search" by NF:\n    "Hey, Nate, how's lif...     nf  
2     "When I Grow Up" by NF:\n    Yeah, when I grow...     nf  
3     "Lie" by NF:\n    I heard you told your friend...     nf  
4     "PAID MY DUES" by NF:\n    Clicks, clicks, cli...     nf  
...                                                 ...    ...  
4023  "Left Somethin’" by Token:\n    [Verse 1]\n   ...  token  
4024  "Back It Up*" by Token:\n    [Chorus]\n    Bac...  token  
4025  "Jurassic Cypher" by Token:\n    [Verse: Token...  token  
4026  "Summer Break" by Token:\n    [Intro]\n    Uh\...  token  
4027  "Yellow Tape" by Token:\n    [Intro]\n    Kato...  token  

[3768 rows x 3 columns]

#### Removing unwanted phrases
I noticed that lyrics had "EmbedShare URLCopyEmbedCopy" plus 1 or more numbers prior to it at the end of the extracted lyrics. Removing those lines and replacing it with a space.

In [30]:
lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace(r"[0-9]*EmbedShare URLCopyEmbedCopy"," ")

<ipython-input-30-cf3ce80320f3>:1: FutureWarning: The default value of regex will change from True to False in a future version.
  lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace(r"[0-9]*EmbedShare URLCopyEmbedCopy"," ")


#### Non-english words

In [31]:
lyrical_df[lyrical_df["lyrics"].str.contains("خۆتان")]

lyrics  \
2060  \nهەڵیبێنم ئەمە چیە، ها؟\nوابزانم ڕوون دیارە\n...   

                                                   song  artist  
2060  "Premonition (Intro) (Kurdish Subtitle)" by Em...  eminem

In [32]:
lyrical_df

lyrics  \
0     Feels like we're on the edge right now\nI wish...   
1     "Hey, Nate, how's life?"\nI don’t know, it's a...   
2     Yeah, when I grow up, you know what I wanna be...   
3     I heard you told your friends that I'm just no...   
4     Clicks, clicks, clicks, they'll do anything to...   
...                                                 ...   
4023  \nToken goin' in, you can call the beat a slot...   
4024  \nBack it up, back it up\nFlow too crack I'm c...   
4025  \nSeptember 24th 1998 was the time that Satan ...   
4026  \nUh\n\n\nIt's summer break, yeah it's summer ...   
4027  \nKato production\n\n\nI don't see no competit...   

                                                   song artist  
0     "Let You Down" by NF:\n    Feels like we're on...     nf  
1     "The Search" by NF:\n    "Hey, Nate, how's lif...     nf  
2     "When I Grow Up" by NF:\n    Yeah, when I grow...     nf  
3     "Lie" by NF:\n    I heard you told your friend...     nf  
4     "PAID MY DUES" by NF:\n    Clicks, clicks, cli...     nf  
...                                                 ...    ...  
4023  "Left Somethin’" by Token:\n    [Verse 1]\n   ...  token  
4024  "Back It Up*" by Token:\n    [Chorus]\n    Bac...  token  
4025  "Jurassic Cypher" by Token:\n    [Verse: Token...  token  
4026  "Summer Break" by Token:\n    [Intro]\n    Uh\...  token  
4027  "Yellow Tape" by Token:\n    [Intro]\n    Kato...  token  

[3768 rows x 3 columns]

In [33]:
lyrical_df = lyrical_df[lyrical_df["lyrics"].map(lambda x: x.isascii())]

In [34]:
lyrical_df

lyrics  \
2     Yeah, when I grow up, you know what I wanna be...   
5     \nHey!\nLeave me alone\n\nPanic-stricken, hand...   
9     I just need some time, I'm tryna think straigh...   
11    How could you leave so unexpected?\nWe waited,...   
12    Insidious is blind inception\nWhat's reality w...   
...                                                 ...   
4020  \nWho do you believe in?\nWho do you believe i...   
4022  Productive, successful, I'm taught like that\n...   
4023  \nToken goin' in, you can call the beat a slot...   
4026  \nUh\n\n\nIt's summer break, yeah it's summer ...   
4027  \nKato production\n\n\nI don't see no competit...   

                                                   song artist  
2     "When I Grow Up" by NF:\n    Yeah, when I grow...     nf  
5     "Leave Me Alone" by NF:\n    (Yeah)\n    Hey!\...     nf  
9     "If You Want Love" by NF:\n    I just need som...     nf  
11    "How Could You Leave Us" by NF:\n    How could...     nf  
12    "Mansion" by NF:\n    Insidious is blind incep...     nf  
...                                                 ...    ...  
4020  "Who Do U Believe In?" by Token:\n    [Intro:]...  token  
4022  "Humble (Remix)" by Token:\n    Productive, su...  token  
4023  "Left Somethin’" by Token:\n    [Verse 1]\n   ...  token  
4026  "Summer Break" by Token:\n    [Intro]\n    Uh\...  token  
4027  "Yellow Tape" by Token:\n    [Intro]\n    Kato...  token  

[2259 rows x 3 columns]

In [35]:
lyrical_df[lyrical_df["lyrics"].str.contains("خۆتان")]

Empty DataFrame
Columns: [lyrics, song, artist]
Index: []

#### Punctuation

In [36]:
# Removing punctuation
lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace("[^\w\s]","")

<ipython-input-36-ba96207683c9>:2: FutureWarning: The default value of regex will change from True to False in a future version.
  lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace("[^\w\s]","")
<ipython-input-36-ba96207683c9>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace("[^\w\s]","")


#### Removing cuss words
Before I filter out the cuss words, I'll force the lyrics column to only contain lowered lettered so my filtering process is easier. That being said, I will not showcase the filtered out words to keep things as professional as possible. I'll just create a blocked out template with the variable **X** indicating the cuss word. 

In [37]:
# lowering 
lyrical_df["lyrics"] = lyrical_df["lyrics"].str.lower()

<ipython-input-37-db3d6a39f587>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  lyrical_df["lyrics"] = lyrical_df["lyrics"].str.lower()


In [38]:
# cuss word filtering
# X = "CUSS WORD"
# lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace(X,"")

In [39]:
# Check
# lyrical_df["lyrics"][1]

Finally done! I hope I've cleaned out all the cuss words out. Time to export this dataframe as a csv and start modelling!

In [74]:
print(lyrical_df.shape)

print(2259/4028)

(2259, 3)
0.560824230387289


Started off with 4028 songs and have reduced the number to 2259. That's a retention of ~56.1%.

In [67]:
lyrical_df

lyrics  \
2     yeah when i grow up you know what i wanna be\n...   
5     \nhey\nleave me alone\n\npanicstricken handle ...   
9     i just need some time im tryna think straight\...   
11    how could you leave so unexpected\nwe waited w...   
12    insidious is blind inception\nwhats reality wi...   
...                                                 ...   
4020  \nwho do you believe in\nwho do you believe in...   
4022  productive successful im taught like that\ngir...   
4023  \ntoken goin in you can call the beat a slot m...   
4026  \nuh\n\n\nits summer break yeah its summer bre...   
4027  \nkato production\n\n\ni dont see no competiti...   

                                                   song artist  
2     "When I Grow Up" by NF:\n    Yeah, when I grow...     nf  
5     "Leave Me Alone" by NF:\n    (Yeah)\n    Hey!\...     nf  
9     "If You Want Love" by NF:\n    I just need som...     nf  
11    "How Could You Leave Us" by NF:\n    How could...     nf  
12    "Mansion" by NF:\n    Insidious is blind incep...     nf  
...                                                 ...    ...  
4020  "Who Do U Believe In?" by Token:\n    [Intro:]...  token  
4022  "Humble (Remix)" by Token:\n    Productive, su...  token  
4023  "Left Somethin’" by Token:\n    [Verse 1]\n   ...  token  
4026  "Summer Break" by Token:\n    [Intro]\n    Uh\...  token  
4027  "Yellow Tape" by Token:\n    [Intro]\n    Kato...  token  

[2259 rows x 3 columns]

#### Blank spaces and lines

In [68]:
lyrical_df

lyrics  \
2     yeah when i grow up you know what i wanna be\n...   
5     \nhey\nleave me alone\n\npanicstricken handle ...   
9     i just need some time im tryna think straight\...   
11    how could you leave so unexpected\nwe waited w...   
12    insidious is blind inception\nwhats reality wi...   
...                                                 ...   
4020  \nwho do you believe in\nwho do you believe in...   
4022  productive successful im taught like that\ngir...   
4023  \ntoken goin in you can call the beat a slot m...   
4026  \nuh\n\n\nits summer break yeah its summer bre...   
4027  \nkato production\n\n\ni dont see no competiti...   

                                                   song artist  
2     "When I Grow Up" by NF:\n    Yeah, when I grow...     nf  
5     "Leave Me Alone" by NF:\n    (Yeah)\n    Hey!\...     nf  
9     "If You Want Love" by NF:\n    I just need som...     nf  
11    "How Could You Leave Us" by NF:\n    How could...     nf  
12    "Mansion" by NF:\n    Insidious is blind incep...     nf  
...                                                 ...    ...  
4020  "Who Do U Believe In?" by Token:\n    [Intro:]...  token  
4022  "Humble (Remix)" by Token:\n    Productive, su...  token  
4023  "Left Somethin’" by Token:\n    [Verse 1]\n   ...  token  
4026  "Summer Break" by Token:\n    [Intro]\n    Uh\...  token  
4027  "Yellow Tape" by Token:\n    [Intro]\n    Kato...  token  

[2259 rows x 3 columns]

In [69]:
# Removing blank lines
lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace("\n"," ")

<ipython-input-69-6a40e9e38a21>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace("\n"," ")


In [70]:
# Removing blank spaces
lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace(r"[ ]{2,}"," ")

<ipython-input-70-0b4be70d2d06>:2: FutureWarning: The default value of regex will change from True to False in a future version.
  lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace(r"[ ]{2,}"," ")
<ipython-input-70-0b4be70d2d06>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  lyrical_df["lyrics"] = lyrical_df["lyrics"].str.replace(r"[ ]{2,}"," ")


In [71]:
lyrical_df

lyrics  \
2     yeah when i grow up you know what i wanna be t...   
5      hey leave me alone panicstricken handle busin...   
9     i just need some time im tryna think straight ...   
11    how could you leave so unexpected we waited we...   
12    insidious is blind inception whats reality wit...   
...                                                 ...   
4020   who do you believe in who do you believe in i...   
4022  productive successful im taught like that girl...   
4023   token goin in you can call the beat a slot ma...   
4026   uh its summer break yeah its summer break the...   
4027   kato production i dont see no competition i s...   

                                                   song artist  
2     "When I Grow Up" by NF:\n    Yeah, when I grow...     nf  
5     "Leave Me Alone" by NF:\n    (Yeah)\n    Hey!\...     nf  
9     "If You Want Love" by NF:\n    I just need som...     nf  
11    "How Could You Leave Us" by NF:\n    How could...     nf  
12    "Mansion" by NF:\n    Insidious is blind incep...     nf  
...                                                 ...    ...  
4020  "Who Do U Believe In?" by Token:\n    [Intro:]...  token  
4022  "Humble (Remix)" by Token:\n    Productive, su...  token  
4023  "Left Somethin’" by Token:\n    [Verse 1]\n   ...  token  
4026  "Summer Break" by Token:\n    [Intro]\n    Uh\...  token  
4027  "Yellow Tape" by Token:\n    [Intro]\n    Kato...  token  

[2259 rows x 3 columns]

### Exporting dataframe into csv

In [72]:
lyrical_df.to_csv("clean_lyrical_df.csv")